### ISSUE TITLE
react-rhythm-of-business-calendar - 

### ISSUE THREAD

**Issue by NaiduKumapatla** [pnp/sp-dev-fx-webparts#3154]
### Disclaimer

Yes

### Sample

react-rhythm-of-business-calendar

### Author(s)

@d-turley

### What happened?

Dates keep moving for all day events every time when the page got refreshed.

### Steps to reproduce

1.Screenshot1: Once the refiners and events are created
2.Screenshot2: After the page is refreshed.
3
![Screenshot1](https://user-images.githubusercontent.com/108256799/202987004-2d4b6ce2-5b6b-4112-80f0-2d40025fcf17.png)

![Screenshot2](https://user-images.githubusercontent.com/108256799/202987011-3854d6ce-6374-4f8f-8546-440de53a1373.png)

### Expected behavior

All events dates are moving except where the time is "defined for all day event".

### Target SharePoint environment

SharePoint Online

### Developer environment

Windows

### Browsers

- [ ] Internet Explorer
- [X] Microsoft Edge
- [X] Google Chrome
- [ ] FireFox
- [ ] Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### What version of Node.js is currently installed on your workstation?

v16.18.1

### What version of Node.js is required by the sample?

16

### Paste the results of SPFx doctor

√ SharePoint Framework v1.15.2
√ Node v16.18.1
√ yo v4.3.0
√ gulp-cli v2.3.0
× typescript v4.5.5 installed in the project

Recommended fixes:

- npm un typescript

### Additional environment details

× typescript v4.5.5 installed in the project
Would not compile unless the above version installed.

**Comment by d-turley** [comment]
Hi @NaiduKumapatla 

Can you provide your system time zone and SPO site time zone settings?

The app has been designed and tested to work across different time zones, however from the screenshots it appears this could be an issue with time zones and caching of data locally in the browser.

Regarding TypeScript, the solution uses the rush stack compiler (@microsoft/rush-stack-compiler-4.5@0.2.2) which has a dependency typescript@4.5.5 and should be sufficient for building the code without needing to install additional packages.

**Comment by martinnaste** [comment]
Hi @d-turley 

I am Martin. I work with Naidu. 

Both time zones would be in UTC+8.

We have found that it there is an issue with the caching yes. We can't pinpoint it on how it works the way it does, but while it is cached, it pushes the event items back a day in the calendar with each refresh. Deleting that cache, resets it back to what it should be upon the next refresh. The cycle starts again from then though.

I believe I had fixed the Typescript issue. But I will revert my changes and install Typescript@4.5.5.
